src/field_theory/splitting_field.lean

```diff
@@ -672,8 +707,31 @@ splitting_field_aux.field _ _
 
 instance inhabited : inhabited (splitting_field f) := ⟨37⟩
 
+/-- This should be an instance globally, but it creates diamonds with the `ℕ` and `ℤ` actions:
+
+```lean
+example :
+  (add_comm_monoid.nat_module : module ℕ (splitting_field f)) =
+    @algebra.to_module _ _ _ _ (splitting_field.algebra' f) :=
+rfl  -- fails
+
+example :
+  (add_comm_group.int_module _ : module ℤ (splitting_field f)) =
+    @algebra.to_module _ _ _ _ (splitting_field.algebra' f) :=
+rfl  -- fails
+```
+
+Until we resolve these diamonds, it's more convenient to only turn this instance on with
+`local attribute [instance]` in places where the benefit of having the instance outweighs the cost.
+
+In the meantime, the `splitting_field.algebra` instance below is safe since `K = ℕ` and `K = ℤ` are
```

Comment:
Did you check `K = ℚ`? Every char 0 field is automatically a `ℚ`-algebra.
```
-- src/algebra/algebra/basic.lean:1211
instance algebra_rat {α} [division_ring α] [char_zero α] : algebra ℚ α :=
```

Reply:
I didn't, because we know there are diamonds in `ℕ`, `ℤ`, and `ℚ` algebras everywhere.

The intent of this comment is to point out that this creates diamonds in `ℕ` and `ℤ` modules, which cannot be solved via the diamond-avoiding mechanism we use elsewhere.

Reply:
Hmmm, then I didn't understand the comment. I thought that it was saying "the instance below is safe, because `K` is a field which `ℕ` and `ℤ` are not, so the annoying diamonds explained above cannot occur".
Could you please clarify the comment if it should mean something else?

Reply:
I guess by "safe" I mean "safe from this particular problem" and "as safe as things were before this PR"

Reply:
I would prefer to rephrase the comment. I agree that the instance is "as safe as things were before this PR", but it isn't "safe". Please add an extra (sub)sentence.

bors d+

Reply:
Does the new subsentence look ok?